x-pack/plugins/inference/server/routes/connectors.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CoreSetup, IRouter, RequestHandlerContext } from '@kbn/core/server';
+import { InferenceConnector, InferenceConnectorType } from '../../common/connectors';
+import type { InferenceServerStart, InferenceStartDependencies } from '../types';
+
+export function registerConnectorsRoute({
+  coreSetup,
+  router,
+}: {
+  coreSetup: CoreSetup<InferenceStartDependencies, InferenceServerStart>;
+  router: IRouter<RequestHandlerContext>;
+}) {
+  router.get(
+    {
+      path: '/internal/inference/connectors',
+      validate: {},
+    },
+    async (_context, request, response) => {
+      const actions = await coreSetup
+        .getStartServices()
+        .then(([_coreStart, pluginsStart]) => pluginsStart.actions);
+
+      const client = await actions.getActionsClientWithRequest(request);
+
+      const allConnectors = await client.getAll({
```

Comment:
@mikecote @ymao1 @pmuellr It would be nice if we could provide to solutions a way to get all connectors filtered by action type IDs so they will not have to get all connectors (max returned is 10K) and then filter out them in-memory. Wdyt?

Reply:
Seems reasonable. I will open an issue.